src/components/CreateTask/CreateTask.tsx

```diff
@@ -0,0 +1,89 @@
+import { Button, Modal, Select, TextInput, Textarea } from "@mantine/core";
+import { DateInput } from "@mantine/dates";
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+import { Task } from "../../types/Task";
+import "@mantine/dates/styles.css";
+import {
+  addTaskToLocalStorage,
+  getLastTaskId,
+  saveLastTaskId,
+} from "../../localstorage/localstorage";
+
+type CreateTaskProps = {
+  opened: boolean;
+  close: () => void;
+};
+
+function CreateTask({ opened, close }: CreateTaskProps) {
+  const { control, handleSubmit, reset } = useForm<Task>();
+  const onSubmit: SubmitHandler<Task> = (data: Task) => {
+    const id = getLastTaskId() + 1;
+    const task: Task = {
+      id: id,
+      title: data.title,
+      description: data.description,
+      dueDate: data.dueDate,
+      priority: data.priority,
+    };
+    addTaskToLocalStorage(task);
+    saveLastTaskId(id);
+    reset();
+  };
+  return (
+    <>
+      <Modal opened={opened} onClose={close} title="Create Task" centered>
+        <form
+          onSubmit={handleSubmit(onSubmit)}
+          style={{ display: "flex", flexDirection: "column", gap: 12 }}
+        >
+          <Controller
+            name="title"
+            control={control}
+            render={({ field }) => (
+              <TextInput label="Task Title" placeholder="title" {...field} />
+            )}
```

Comment:
None of the fields contain validation. Please create a new ticket to address form validation.

Reply:
Created Ticket for form validation. link: https://trello.com/c/YFdHtcuq